src/PublicApiGenerator/MethodNameBuilder.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.CodeDom;
+using System.Linq;
+using Mono.Cecil;
+using static System.String;
+
+namespace PublicApiGenerator
+{
+    public static class MethodNameBuilder
+    {
+        public static string AugmentMethodNameWithMethodModifierMarkerTemplate(MethodDefinition methodDefinition,
+            MemberAttributes attributes)
+        {
+            var name = CSharpOperatorKeyword.Get(methodDefinition.Name);
+
+            if (methodDefinition.DeclaringType.IsInterface)
+            {
+                return name;
+            }
+
+            bool? isNew = null;
+            var baseType = methodDefinition.DeclaringType.BaseType;
+            while (baseType != null)
+            {
+                var typeDef = baseType as TypeDefinition;
+                // too simple, improve
+                isNew = typeDef?.Methods.Any(e => e.Name.Equals(methodDefinition.Name, StringComparison.Ordinal) && e.Parameters.Count == methodDefinition.Parameters.Count);
```

Comment:
`e.Name.Equals(methodDefinition.Name, StringComparison.Ordinal)` does the same comparison as `e.Name == methodDefinition.Name`